DigitalLearningSolutions.Data.Tests/Services/CourseServiceTests.cs

```diff
@@ -244,13 +240,32 @@ public void UpdateLearningPathwayDefaultsForCourse_calls_data_service()
                 .MustHaveHappened();
         }
 
+        [Test]
+        public void GetAllCoursesForDelegate_returns_only_courses_at_centre()
+        {
+            // Given
+            const int delegateId = 1;
+            const int centreId = 1;
+            const int categoryId = 1;
+            var delegateCourseInfoAtCentre = new DelegateCourseInfo { CustomisationCentreId = centreId, CourseCategoryId = categoryId };
+            var delegateCourseInfoNotAtCentre = new DelegateCourseInfo { CustomisationCentreId = 1000, CourseCategoryId = categoryId };
+            A.CallTo(() => courseDataService.GetDelegateCoursesInfo(delegateId))
+                .Returns(new[] { delegateCourseInfoAtCentre, delegateCourseInfoNotAtCentre });
+
+            // When
+            var result = courseService.GetAllCoursesInCategoryForDelegate(delegateId, centreId, categoryId);
+
+            // Then
+            result.Count().Should().Be(1);

```

Comment:
I think we should also check that the result's returned model should have the correct centre Id.